Now, let's finally look at that divorced dataset.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('../data/divorces_2000-2015_translated.csv')
print(df.shape)
df.columns = df.columns.str.lower()
df.columns = df.columns.str.replace(' ', '_')
df.columns = df.columns.str.replace('.', '')
df.head()

(4923, 41)


divorce_date type_of_divorce nationality_partner_man dob_partner_man  \
0       9/6/06       Necesario                MEXICANA        18/12/75   
1       1/2/00      Voluntario                MEXICANA             NaN   
2       1/2/05       Necesario                MEXICANA         22/2/55   
3       1/2/06       Necesario                MEXICANA         20/1/64   
4       1/2/06       Necesario                MEXICANA        30/10/75   

  place_of_birth_partner_man birth_municipality_of_partner_man  \
0          XALAPA - ENRIQUEZ                            XALAPA   
1                        NaN                               NaN   
2          XALAPA - ENRIQUEZ                            XALAPA   
3          XALAPA - ENRIQUEZ                            XALAPA   
4          XALAPA - ENRIQUEZ                            XALAPA   

  birth_federal_partner_man birth_country_partner_man  age_partner_man  \
0                  VERACRUZ                    MEXICO             30.0   
1                       NaN                       NaN             47.0   
2                  VERACRUZ                    MEXICO             49.0   
3                  VERACRUZ                    MEXICO             42.0   
4                  VERACRUZ                    MEXICO             30.0   

  residence_municipality_partner_man  ... marriage_certificate_municipality  \
0                             XALAPA  ...                            XALAPA   
1                                NaN  ...                            XALAPA   
2                                NaN  ...                            XALAPA   
3                             XALAPA  ...                            XALAPA   
4                           COATEPEC  ...                            XALAPA   

  marriage_certificate_federal  level_of_education_partner_man  \
0                     VERACRUZ                      SECUNDARIA   
1                     VERACRUZ                    PREPARATORIA   
2                     VERACRUZ                    PREPARATORIA   
3                     VERACRUZ                     PROFESIONAL   
4                     VERACRUZ                     PROFESIONAL   

  employment_status_partner_man level_of_education_partner_woman  \
0                        OBRERO                       SECUNDARIA   
1               ESTABLECIMIENTO                     PREPARATORIA   
2                        OBRERO                              NaN   
3                      EMPLEADO                     PREPARATORIA   
4                      EMPLEADO                     PREPARATORIA   

                employment_status_partner_woman marriage_duration  \
0                                      EMPLEADO               5.0   
1                                      EMPLEADO               NaN   
2  TRABAJADOR POR CUENTA PROPIA EN VIA PUBLICA                NaN   
3                                      EMPLEADO              18.0   
4                                    NO TRABAJA               7.0   

  marriage_duration_months num_children custody  
0                      NaN          1.0     NaN  
1                      NaN          NaN     NaN  
2                      NaN          NaN     NaN  
3                      NaN          2.0   MADRE  
4                      NaN          2.0   MADRE  

[5 rows x 41 columns]

Yeah, once again we need to do a lot of sprucing. We'll definitely err on the side of making things as simple as possible.

In [3]:
 i = 0

In [4]:
df[df.columns[i]].value_counts() #Yeah, we don't care about this... way too detailed for our purposes. Do note though
#the range of them

divorce_date
9/10/09     9
5/9/05      8
25/2/09     8
21/11/03    8
3/11/04     8
           ..
8/10/13     1
8/10/14     1
8/11/01     1
8/11/06     1
31/12/13    1
Name: count, Length: 2596, dtype: int64

In [5]:
min(df['divorce_date']), max(df['divorce_date'])

('1/10/01', '9/9/14')

In [6]:
to_be_deleted = []

to_be_deleted.append(df.columns[i])
print(f"Next up to be deleted is {df.columns[i]} and now we're poised to delete a total of {len(to_be_deleted)} from our dataframe.")

Next up to be deleted is divorce_date and now we're poised to delete a total of 1 from our dataframe.


In [7]:
i += 1
df[df.columns[i]].value_counts()

type_of_divorce
Necesario     2528
Voluntario    2395
Name: count, dtype: int64

Guess we'll worry about one-hot-encodng later.

Pragmatically, I'd assume Necesario should be 0. Ie the type of scenario we'd ideally like to avoid, but relatively 'better' than the alternative (in short...).

In [8]:
i += 1
df[df.columns[i]].value_counts()

nationality_partner_man
MEXICANA          4879
CUBANA               8
ESTADOUNIDENSE       7
ESPAÑOLA            6
ARGENTINA            4
FRANCESA             2
CANADIENSE           2
ALEMANA              2
VENEZOLANA           2
JAPONESA             1
AUSTRALIANA          1
POLACA               1
CHINA                1
ITALIANA             1
CHILENA              1
COSTARRICENSE        1
AUSTRIACA            1
COLOMBIANA           1
NICARAGUENSE         1
Name: count, dtype: int64

Yeah, we'll simplify this to mexican/not. Although we'd be interested to see a mexican of any type with a non, we're also interested in a case where neither are mexican yet are still living there (if it occurs in this dataset).

In [9]:
df[df.columns[i]] = df[df.columns[i]].apply(lambda x: 0 if x == 'MEXICANA' else 1)

In [10]:
df.head()

divorce_date type_of_divorce  nationality_partner_man dob_partner_man  \
0       9/6/06       Necesario                        0        18/12/75   
1       1/2/00      Voluntario                        0             NaN   
2       1/2/05       Necesario                        0         22/2/55   
3       1/2/06       Necesario                        0         20/1/64   
4       1/2/06       Necesario                        0        30/10/75   

  place_of_birth_partner_man birth_municipality_of_partner_man  \
0          XALAPA - ENRIQUEZ                            XALAPA   
1                        NaN                               NaN   
2          XALAPA - ENRIQUEZ                            XALAPA   
3          XALAPA - ENRIQUEZ                            XALAPA   
4          XALAPA - ENRIQUEZ                            XALAPA   

  birth_federal_partner_man birth_country_partner_man  age_partner_man  \
0                  VERACRUZ                    MEXICO             30.0   
1                       NaN                       NaN             47.0   
2                  VERACRUZ                    MEXICO             49.0   
3                  VERACRUZ                    MEXICO             42.0   
4                  VERACRUZ                    MEXICO             30.0   

  residence_municipality_partner_man  ... marriage_certificate_municipality  \
0                             XALAPA  ...                            XALAPA   
1                                NaN  ...                            XALAPA   
2                                NaN  ...                            XALAPA   
3                             XALAPA  ...                            XALAPA   
4                           COATEPEC  ...                            XALAPA   

  marriage_certificate_federal  level_of_education_partner_man  \
0                     VERACRUZ                      SECUNDARIA   
1                     VERACRUZ                    PREPARATORIA   
2                     VERACRUZ                    PREPARATORIA   
3                     VERACRUZ                     PROFESIONAL   
4                     VERACRUZ                     PROFESIONAL   

  employment_status_partner_man level_of_education_partner_woman  \
0                        OBRERO                       SECUNDARIA   
1               ESTABLECIMIENTO                     PREPARATORIA   
2                        OBRERO                              NaN   
3                      EMPLEADO                     PREPARATORIA   
4                      EMPLEADO                     PREPARATORIA   

                employment_status_partner_woman marriage_duration  \
0                                      EMPLEADO               5.0   
1                                      EMPLEADO               NaN   
2  TRABAJADOR POR CUENTA PROPIA EN VIA PUBLICA                NaN   
3                                      EMPLEADO              18.0   
4                                    NO TRABAJA               7.0   

  marriage_duration_months num_children custody  
0                      NaN          1.0     NaN  
1                      NaN          NaN     NaN  
2                      NaN          NaN     NaN  
3                      NaN          2.0   MADRE  
4                      NaN          2.0   MADRE  

[5 rows x 41 columns]

In [11]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 3.


dob_partner_man
25/5/73     5
5/4/74      5
17/3/75     4
23/4/74     4
2/8/61      4
           ..
1/10/72     1
19/6/68     1
5/3/70      1
20/12/77    1
7/2/58      1
Name: count, Length: 3848, dtype: int64

Hmm, potentially let us simplify.... Oh wait, I just ealized - I want to example their relative ages.Hence, I'd need tha divocre data ultimately.... we'll keep on going for now though.

In [12]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 4.


place_of_birth_partner_man
XALAPA - ENRIQUEZ    2472
DISTRITO FEDERAL      347
VERACRUZ              129
COATEPEC               68
POZA RICA              49
                     ... 
ANTONIO HIDALGO         1
ESPARTACO               1
AHUACATAN               1
SAN PABLO COAPAN        1
EL LIMON                1
Name: count, Length: 669, dtype: int64

Yeah, we're simplifyng this. Let the cut off be 10 (beyond the scope to comment here why I'm chosing that number).

In [13]:
to_be_deleted.append(df.columns[i])
print(f"Next up to be deleted is {df.columns[i]} and now we're poised to delete a total of {len(to_be_deleted)} from our dataframe.")

Next up to be deleted is place_of_birth_partner_man and now we're poised to delete a total of 2 from our dataframe.


In [14]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 5.


birth_municipality_of_partner_man
XALAPA                   2479
DISTRITO FEDERAL          348
VERACRUZ                  131
COATEPEC                   78
MARTINEZ DE LA TORRE       60
                         ... 
PATZCUARO                   1
TOPILITOS DE ZARAGOZA       1
TULA DE ALLENDE             1
PARRAL                      1
ZAUTLA                      1
Name: count, Length: 424, dtype: int64

Somehow different??? A bit tempted to delete as a quick scan shows barely any differencs. However, this seems more inclusive so I'll go with this (424 vs. 669).

In [15]:
list(df[df.columns[i]].value_counts(normalize=True)[:9].index)

['XALAPA',
 'DISTRITO FEDERAL',
 'VERACRUZ',
 'COATEPEC',
 'MARTINEZ DE LA TORRE',
 'MISANTLA',
 'POZA RICA',
 'ORIZABA',
 'CORDOBA']

In [16]:
print(sum(df[df.columns[i]].value_counts(normalize=True)[:9]))

0.6846057571964955


In [17]:
#I feel like I had a neat code to do this. Oh Well.

feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:9].index)

#Somewhat arbitrarily chosen, but eh. After like 7 they have such a small amount... Will simplify.

df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [19]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 6.


birth_federal_partner_man
VERACRUZ              3920
DISTRITO FEDERAL       403
PUEBLA                 111
TAMAULIPAS              33
OAXACA                  30
                      ... 
EL SALVADOR              1
GALICIA                  1
MISSOURI                 1
CAMAG√úEY                1
NORTE DE SANTANDER       1
Name: count, Length: 76, dtype: int64

Similar to the other.... We might as well keep it Especially since we're condensing. This time we'll simplify it to Puebla than 'Other' as they don' even have 100...

In [21]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:3].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [22]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 7.


birth_country_partner_man
MEXICO                       4748
CUBA                            8
ESPAÑA                         8
ESTADOS UNIDOS DE AMERICA       6
ARGENTINA                       3
COLOMBIA                        3
FRANCIA                         3
VENEZUELA                       2
POLONIA                         2
ALEMANIA                        2
COSTA RICA                      1
CANADA                          1
CENTRO AMERICA                  1
CHILE                           1
IRAN                            1
ITALIA                          1
JAPON                           1
MARYLAND                        1
INGLATERRA                      1
AUSTRALIA                       1
NICARAGUA                       1
Name: count, dtype: int64

Unsurprising... we'll keep both for now. Especially when this will get simplified to binary Mexico or not.

In [23]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:1].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [25]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 8.


age_partner_man
34.0    235
31.0    216
35.0    212
32.0    203
36.0    203
       ... 
78.0      2
76.0      2
75.0      2
91.0      1
83.0      1
Name: count, Length: 62, dtype: int64

I just reconfirmed and yes this is referring to dod. Hence, we're good to go to simplify by removing (as we originally assumed) the dob. However, I'd still be interested in seeing about how old they were when they got married. So, we'll see.

In [27]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 9.


residence_municipality_partner_man
XALAPA                 3927
VERACRUZ                 67
COATEPEC                 63
EMILIANO ZAPATA          62
DISTRITO FEDERAL         40
                       ... 
SEBASTIAN TUXTLA          1
GUSTAVO A. MADERO         1
PETLALCINGO               1
TECOLUTILLA               1
SANTA MARIA EL TULE       1
Name: count, Length: 175, dtype: int64

In [29]:
df[df.columns[i]].value_counts(normalize=True)[:4]

residence_municipality_partner_man
XALAPA             0.853881
VERACRUZ           0.014568
COATEPEC           0.013699
EMILIANO ZAPATA    0.013481
Name: proportion, dtype: float64

Eh... who cares. We'll keep it. Simplify it to anything past Emiliana Zapata ak 1 percent is other.

In [30]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:4].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [32]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()[:5]

Now looking at feature 11.


residence_country_partner_man
MEXICO                       4589
ESTADOS UNIDOS DE AMERICA       7
ESPAÑA                         1
AUSTRALIA                       1
PUERTO RICO                     1
Name: count, dtype: int64

Yeah, this is becoming binary. More spread out than the 'birth_country_partner_man' so worth having both.

In [33]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:1].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [34]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 12.


monthly_income_partner_man_peso
5000.0     326
3000.0     301
4000.0     279
10000.0    276
6000.0     275
          ... 
5900.0       1
6600.0       1
4900.0       1
6638.0       1
5.0          1
Name: count, Length: 190, dtype: int64

Heads up already re. potential inflation issues. However, the data isn't too far apart so we'll likely be lax and not bother for the sake of this study.

Perhaps if it turns out that this feature is super significant we'll reconsider.

In [36]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

Now looking at feature 13.


occupation_partner_man
EMPLEADO                2352
COMERCIANTE              299
PROFESOR                 132
MAESTRO                   89
MEDICO                    87
                        ... 
AGENTE DE SEGUROS          1
PRODUCTOR                  1
ADMINISTRATIVO             1
MEDICO ANESTECIOLOGO       1
CAPITAN DE CORBETA         1
Name: count, Length: 221, dtype: int64

Hmm,pretty small after the first one. Potentially a move to just make this binary. At least for now we'll conntinue.

First 4 are more than 2%; 13 are 1% so at most do 13 than an other category. They sum up to 65 and 80%.

In [44]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts(normalize=True)

Now looking at feature 14.


place_of_residence_partner_man
XALAPA-ENRIQUEZ     3918
VERACRUZ              66
COATEPEC              57
EMILIANO ZAPATA       37
DISTRITO FEDERAL      34
                    ... 
ZONGOLICA              1
CERRO COLORADO         1
MIGUEL HIDALGO         1
ALVARO OBREGON         1
LAS PUENTES            1
Name: count, Length: 229, dtype: int64

Similar to things we've seen before, but eh...

In [45]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:3].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

## Re. the woman.

By default won't bother to comment unless I notice trends that I find significant between the two. Ie even if the nationality is much more likely to be Mexican if a woman I won't bother commenting given that they're similar.

In [46]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts(normalize=True)

Now looking at feature 15.


nationality_partner_woman
MEXICANA          0.993902
CUBANA            0.001220
ESTADOUNIDENSE    0.001016
FRANCESA          0.000610
ESPAÑOLA         0.000610
ARGENTINA         0.000407
POLACA            0.000407
BRASILEÑA        0.000203
PERUANA           0.000203
DOMINICANA        0.000203
VENEZOLANA        0.000203
COLOMBIANA        0.000203
BOLIVIANA         0.000203
JAPONESA          0.000203
BRITANICA         0.000203
SUIZA             0.000203
Name: proportion, dtype: float64

In [47]:
feature_keepers = list(df[df.columns[i]].value_counts(normalize=True)[:1].index)
df[df.columns[i]] = ["OTHER" if i not in feature_keepers  else i for i in df[df.columns[i]]]

In [ ]:
i += 1
print(f"Now looking at feature {i}.")
df[df.columns[i]].value_counts()

In [ ]:
stopstop

OHE:

type_of_divorce

?bucket monthly incomes?

occupation
